packages/opentelemetry-plugin-user-interaction/src/userInteraction.ts

```diff
@@ -0,0 +1,457 @@
+    const plugin = this;
+    return (original: any) => {
+      return function patchScheduleTask<T extends Task>(
+        this: Zone,
+        task: AsyncTask
+      ) {
+        const currentZone = Zone.current;
+        const currentSpan: types.Span = currentZone.get(ZONE_SCOPE_KEY);
+        if (currentSpan && plugin._shouldCountTask(task, currentZone)) {
+          plugin._incrementTask(currentSpan);
+          plugin._checkForTimeout(task, currentSpan);
+        }
+        return original.call(this, task) as T;
+      };
+    };
+  }
+
+  /**
+   * Patches zone run task - this is done to be able to create a span when
+   * user interaction starts
+   * @private
+   */
+  private _patchZoneRunTask() {
+    const plugin = this;
+    return (original: RunTaskFunction): RunTaskFunction => {
+      return function patchRunTask(
+        this: Zone,
+        task: AsyncTask,
+        applyThis?: any,
+        applyArgs?: any
+      ): Zone {
+        const target: HTMLElement | undefined = plugin._getClickedElement(task);
+        let span: types.Span | undefined;
+        if (target) {
+          span = plugin._createSpan(target, 'click');
+          if (span) {
+            plugin._incrementTask(span);
+            try {
+              return plugin._tracer.withSpan(span, () => {
+                const currentZone = Zone.current;
+                task._zone = currentZone;
+                return original.call(currentZone, task, applyThis, applyArgs);
+              });
+            } finally {
+              plugin._decrementTask(span);
+            }
+          }
+        } else {
+          span = this.get(ZONE_SCOPE_KEY);
+        }
+
+        try {
+          return original.call(this, task, applyThis, applyArgs);
+        } finally {
+          if (span && plugin._shouldCountTask(task, Zone.current)) {
+            plugin._decrementTask(span);
+          }
+        }
+      };
+    };
+  }
+
+  /**
+   * Decides if task should be counted.
+   * @param task
+   * @param currentZone
+   * @private
+   */
+  private _shouldCountTask(task: AsyncTask, currentZone: Zone): boolean {
+    if (task._zone) {
+      currentZone = task._zone;
+    }
+    if (!currentZone || !task.data || task.data.isPeriodic) {
+      return false;
+    }
+    const currentSpan = currentZone.get(ZONE_SCOPE_KEY);
+    if (!currentSpan) {
+      return false;
+    }
+    if (!this._uiMem.get(currentSpan)) {
+      return false;
+    }
+    return task.type === 'macroTask' || task.type === 'microTask';
+  }
+
+  /**
+   * Will try to end span when such span still exists.
+   * @param span
+   * @param endTime
+   * @private
+   */
+  private _tryToEndSpan(span: types.Span, endTime?: types.HrTime) {
+    if (span) {
+      const uiMem = this._uiMem.get(span);
+      if (uiMem) {
+        span.end(endTime);
+        this._uiMem.delete(span);
+      }
+    }
+  }
+
+  /**
+   * implements patch function
+   */
+  protected patch() {
+    const _window = window as any;
```

Comment:
Would it be possible to create an interface type with the specific variables you expect to be getting from `window` (e.g. `WindowWithZone`) and cast it to that instead of casting it to `any`?

That way you are being explicit about what you expect to be present and will help make the code more type-safe if someone e.g. mis-spelled a variable name, etc.